src/components/shapes/attributes.js

```diff
@@ -7,19 +7,54 @@ var dash = require('../drawing/attributes').dash;
 var extendFlat = require('../../lib/extend').extendFlat;
 var templatedArray = require('../../plot_api/plot_template').templatedArray;
 var axisPlaceableObjs = require('../../constants/axis_placeable_objects');
+var basePlotAttributes = require('../../plots/attributes');
 var shapeTexttemplateAttrs = require('../../plots/template_attributes').shapeTexttemplateAttrs;
 var shapeLabelTexttemplateVars = require('./label_texttemplate');
 
 module.exports = templatedArray('shape', {
-    visible: {
+    visible: extendFlat({}, basePlotAttributes.visible, {
+        editType: 'calc+arraydraw'
```

Comment:
`description` may want to be adjusted too, as it refers to `trace` a couple of times.

Reply:
true of several other `basePlotAttributes` you're reusing here as well.

Reply:
Good call. Addressed in [8a0f60a](https://github.com/plotly/plotly.js/pull/6653/commits/8a0f60a4ae24eb5f7839e0f79279e8ea20c557a0).